lua/autorun/server/sv_hideholos.lua

```diff
@@ -1,15 +1,29 @@
+util.AddNetworkString( "update_hologram_visibility_preference" )
+util.AddNetworkString( "get_user_hologram_visibility_preference" )
+
 -- Updates the visibility of all holos for a specific player
 local function updateHoloVisibility( ply )
     for _, holo in pairs( ents.FindByClass( "gmod_wire_hologram" ) ) do
         holo:SetPreventTransmit( ply, ply.cfc_holosDisabled )
     end
+
+    net.Start( "update_hologram_visibility_preference" )
+        net.WriteBool( ply.cfc_holosDisabled )
+    net.Send( ply )
 end
 
--- Initialize the player's cfc_holosDisabled variable and hide all existing holos
-local function initializePlayer( ply )
-    ply.cfc_holosDisabled = true
+net.Receive( "get_user_hologram_visibility_preference", function( len, ply )
+    ply.cfc_holosDisabled = net.ReadBool()
 
     updateHoloVisibility( ply )
+end )
+
+-- Initialize the player's cfc_holosDisabled variable and hide all existing holos
+local function initializePlayer( ply )
+    timer.Create( 5, function()
+        net.Start( "get_user_hologram_visibility_preference" )
+        net.Send( ply )
+    end )
 end
```

Comment:
Maybe we could hook `Tick` or `Think` instead of using a timer here?

Reply:
How do you mean...?

Reply:
Instead of hooking `PlayerInitialSpawn` (which as I understand, can be called before the player is actually fully loaded), we could hook on to `Tick` or `Think` which I believe only run once the player is fully in.

So you could do
```lua
hook.Add("Think", "GetHoloPrefs", function()
    net.Start(blah)
    net.Send()
    hook.Remove("Think", "GetHoloPrefs")
end
```

Reply:
`Think` is every frame, `Tick` is every tick. This would get the preferences every frame if we used `Think`.

Reply:
The idea is you remove the hook immediately after it's called for the first time

Reply:
I don't think that would be as reliable as you might think; when I join I'm able to move for a short period of time before gmod stops me for a short period to finish up whatever it needs to do. The problem was that players would crash/time out if messages were received and/or sent while they are in this period.

Reply:
Well `PlayerInitialSpawn` actually happens before the player is physically in the server - at least on `Think` we're sure they exist in the server. With a Timer, you're only right _most_ of the time